kamu-core/src/infra/repos/object_repository_local_fs.rs

```diff
@@ -38,13 +40,14 @@ where
     D: Send + Sync,
     D: digest::Digest,
 {
-    pub fn new<P>(root: P) -> Self
+    pub fn new<P>(root: P, external_base_url: Option<Url>) -> Self
```

Comment:
If we put `LocalDatasetRepositoryS3` in charge of S3 external URLs then base URL will not appear anywhere but in `ObjectRepositoryS3`. All others would simply return `Err(GetDownloadUrlError::NotSupported)` making protocol to fall back to proxying traffic.

Reply:
I do not understand what do you mean by "fall back to proxying traffic". We have already started a websocket-based communication loop that has different rules. What I read in this comment looks like something that we did not define in the smart transfer protocol description at all..

Reply:
Who would inject this new dataset repository into the catalog? We don't have anything else on today except our CLI app. What you are describing here, is that "platform" would have it's own catalog version, and that would inject different dataset repository. So, if I still use CLI app as the ground, this means these download strategies I tried to implement will never even enable themselves.